src/org/exist/client/InteractiveClient.java

```diff
@@ -2213,23 +2086,22 @@ private void connectToDatabase() {
         try {
             connect();
         } catch (final Exception cnf) {
-            if (startGUI && frame != null) {
+            if (options.startGUI && frame != null) {
                 frame.setStatus("Connection to database failed; message: " + cnf.getMessage());
             } else {
                 System.err.println("Connection to database failed; message: " + cnf.getMessage());
             }
             cnf.printStackTrace();
-            System.exit(1);
+            System.exit(SystemExitCodes.CATCH_ALL_GENERAL_ERROR_EXIT_CODE);
         }
     }
 
     /**
      * Main processing method for the InteractiveClient object
      *
-     * @param args arguments from main()
      * @return true on success, false on failure
      */
-    public boolean run(final String args[]) throws Exception {
+    public boolean run() throws Exception {
```

Comment:
This code was destined for eXist 3.0 but got left behind due to the "outrage about" Guava.

I think we can use some common sense here. As we know eXist does not have a proper public API, and if it did this certainly wouldn't be part of it. This method is called from `main` and it makes no sense to call it from elsewhere, it's a legacy mistake that it is not `private`.